azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunner2Test.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.execapp;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import azkaban.dag.Dag;
+import azkaban.dag.DagBuilder;
+import azkaban.dag.DagProcessor;
+import azkaban.dag.DagService;
+import azkaban.dag.Node;
+import azkaban.dag.NodeProcessor;
+import azkaban.dag.Status;
+import azkaban.project.NodeBean;
+import azkaban.project.NodeBeanLoader;
+import azkaban.utils.ExecutorServiceUtils;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.Test;
+
+/**
+ * Tests for running flows.
+ */
+public class FlowRunner2Test {
+
+  private final DagService dagService = new DagService(new ExecutorServiceUtils());
+  private final CountDownLatch flowFinishedLatch = new CountDownLatch(1);
+
+  // The recorded event sequence.
+  private final List<String> eventSequence = new ArrayList<>();
+
+  @Test
+
+  public void runSimpleV2Flow() throws Exception {
+    final NodeBean flowNode = loadFlowNode();
+    final Dag dag = createDag(flowNode);
+    this.dagService.startDag(dag);
+    this.flowFinishedLatch.await(2, TimeUnit.SECONDS);
+    assertThat(this.eventSequence).isEqualTo(Arrays.asList("n1", "n2"));
+    this.dagService.shutdownAndAwaitTermination();
+  }
+
+  private NodeBean loadFlowNode() throws Exception {
+    final File flowFile = loadFlowFileFromResource();
+    final NodeBeanLoader beanLoader = new NodeBeanLoader();
+    return beanLoader.load(flowFile);
+  }
+
+  private Dag createDag(final NodeBean flowNode) {
+    final DagCreator creator = new DagCreator(flowNode);
+    return creator.create();
+
+  }
+
+  private class DagCreator {
+
+    private final NodeBean flowNode;
+    private final DagBuilder dagBuilder;
+
+    DagCreator(final NodeBean flowNode) {
+      final String flowName = flowNode.getName();
+      this.flowNode = flowNode;
+      this.dagBuilder = new DagBuilder(flowName, new SimpleDagProcessor());
+    }
+
+    Dag create() {
```

Comment:
It seems to me that logic of creating Dag is similar to what has been done in DirectoryYamlFlowLoader. Would it be easier to just reuse the existing part? Do you plan to eventually integrate them together or keep them as separate paths?

Reply:
I haven't looked at the DirectoryYamlFlowLoader code in detail.

Eventually, yes, we will need to handle running both v1 and v2 flows. Will look into reducing logic duplication then.